### PR TITLE
[MP-156] QPP Encryption

### PR DESCRIPTION
**Pull request by danny-cpp**
Draft pull to address bug issue

### CODE REVIEW

File: C++/src/Driver.cpp [hunk 1]
```diff
@@ -1,20 +1,29 @@
...
     std::cout << result << std::endl;
+}
+int main() {
+
+    std::cout << "Hello World" << std::endl;
+
+    // AES_test();
+
+    std::vector<int> seed = {1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16};
+    QPP::QuantumPermutationPad qpp(seed);
```
Comment (danny-cpp) [inline C++/src/Driver.cpp]: Create using pointer via QPP....* a = new QPP... . Exceed stack limit here that's why there is a seg fault

File: C++/src/cryptography/QuantumPermutationPad.cpp [hunk 2]
```diff
@@ -66,22 +91,89 @@ namespace QPP {
...
+            this->cipher_text_vector.push_back(gate.multiply(each));
+        }
         
     }
     
     // Generate a plain text messsage column vector. This involves an XOR with
     // n plaintext bits with n random bits from the prng. The result should be
     // mapped to one column vector from the computational basis.
     void QuantumPermutationPad::generateVector(std::string plain_text) {
+        for (int i = 0; i < sizeof(plain_text); i+=params::mat_len) {
```
Comment (danny-cpp) [inline C++/src/cryptography/QuantumPermutationPad.cpp]: !!!Critical bug!!! Use plain_text.size(). Can u figure out what sizeof() return?

File: C++/src/cryptography/QuantumPermutationPad.cpp [hunk 3]
```diff
@@ -66,22 +91,89 @@ namespace QPP {
...
+            if (plain_text[i] == '\n' || plain_text[i] == '\0'){
```
Comment (danny-cpp) [inline C++/src/cryptography/QuantumPermutationPad.cpp:154]: Can't index letter like this, not Python.

File: C++/src/cryptography/QuantumPermutationPad.cpp [hunk 4]
```diff
@@ -5,20 +5,33 @@ namespace QPP {
...
-        if(seed.size() != (params::M * params::n * params::mat_len)) {
-            exit(1);
-        }
-        this->seed = seed;
-
-        // Create M permutation matrices
-        for(int i = 0; i < params::M; i++) {
-            generateMatrix(i);
-        }
+       if(seed.size() != (params::M * params::n * params::mat_len)) {
```
Comment (danny-cpp) [inline C++/src/cryptography/QuantumPermutationPad.cpp:8]: Print out problem before exit

Comment (lwt0000) [reply C++/src/cryptography/QuantumPermutationPad.cpp:8]: resolved

File: C++/src/cryptography/QuantumPermutationPad.cpp (hunk 2 above)
Comment (lwt0000) [reply C++/src/cryptography/QuantumPermutationPad.cpp]: resolved

File: C++/src/Driver.cpp (hunk 1 above)
Comment (lwt0000) [reply C++/src/Driver.cpp]: resolved

Comment (lwt0000) [reply C++/src/Driver.cpp]: Still have seg falut

**Pull request closed without merging**